src/schematools/types.py

```diff
@@ -989,12 +989,24 @@ def get_dimension_fieldnames(self) -> Dict[str, Tuple[str, str]]:
     def sub_fields(self) -> Iterator[DatasetFieldSchema]:
         """Return the sub fields for a nested structure.
 
+        Calls the `get_sub_fields` method without argument,
+        so no prefixes are added to the field ids.
+        This is the default situation.
+        """
+        return self.get_sub_fields()
```

Comment:
```suggestion
    def sub_fields(self) -> List[DatasetFieldSchema]:
        """Return the sub fields for a nested structure.

        Calls the `get_sub_fields` method without argument,
        so no prefixes are added to the field ids.
        This is the default situation.
        """
        return list(self.get_sub_fields())
```

and make it a `@cached_property`

Reply:
Good idea, gets called quite often.